src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -467,6 +474,24 @@ def __should_run_history(self) -> bool:
             logging.error("Failed to check run history condition", exc_info=True)
             return False
 
+    def __discovery_watchdog(self):
+        logging.info("Cluster discovery watchdog initialized")
+        is_initial_health_check = True
+        while self.__active:
+            # if the initial thread is stuck it will register as "Healthy"
+            initial_discovery_stuck = not is_initial_health_check and self.last_send_time == 0
+            # if is not healthy we will get the stack dump from sigint handler
+            if initial_discovery_stuck:
+                StackTracer.dump(traces=1)
```

Comment:
I think we always want few dumps.
Other wise you might not be able to tell if the thread is stuck (if for example, there's a weird scenario of an endless loop)